scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -827,7 +831,18 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    */
   def materialize: Future[Tap[T]] = {
     val filename = "scio-materialize-" + UUID.randomUUID().toString
-    val tmpDir = context.options.getTempLocation
+    val tmpDir = if (context.options.getTempLocation != null) {
+      context.options.getTempLocation
+    } else {
+      val m = "Specify a temporary location via --tempLocation or PipelineOptions.setTempLocation."
```

Comment:
For locally runner that is already the case, but for other runners was thinking of using reusing gcp temp location or throwing meaningful error like in this PR.